graphql-native/src/main/java/io/ballerina/stdlib/graphql/runtime/engine/CallableUnitCallback.java

```diff
@@ -120,22 +121,33 @@ static void getDataFromService(Environment environment, BObject service, BObject
         }
     }
 
-    static void getDataFromArray(BObject node, BArray result, BMap<BString, Object> data) {
+    static void getDataFromArray(Environment environment, BObject visitor, BObject node, BArray result, BMap<BString,
+            Object> data) {
         if (isScalarType(result.getElementType())) {
             data.put(node.getStringValue(NAME_FIELD), result);
+        } else if (result.getElementType().getTag() == TypeTags.SERVICE_TAG) {
```

Comment:
This check should be done in the `getDataFromResult` method. Otherwise, we are duplicating the array iteration separately for Service types and the other types. The better way should be to call the `getDataFromService` inside the `getDataFromResult` method.